src/app/space/create/codebases/codebases.component.ts

```diff
@@ -268,6 +274,7 @@ export class CodebasesComponent implements OnDestroy, OnInit {
           this.codebases.unshift({} as Codebase); // Add empty object for row header
           this.applyFilters(this.appliedFilters);
         }
+        this.startIdleChe();
```

Comment:
I understand why we're starting Che here, but it's too late at this point. 

We call updateCodebases when the codebaseDeleted event is receieved. And if we cannot update, we call startIdelChe. However,  updateCodebases will never be called again -- the codebases list won't be updated.

We may need to test if Che is running before calling updateCodebases? Or, add something in the page to ping Che periodically and keep it alive?

Reply:
startIdleChe - fetches the Che status and propagates it to other components (if status is not running - starts che). Otherwise, now any update of the list of codebases (add or delete) leads to a bug , when workspaces component and menu - are not displayed, though the che is running, cause all components were reseted to default state - (cheRunning - false). What can be done here for the code to be more explicit - call method smth like - fetchCheState/Status.
Attaching screenshot of the screen after adding new codebase:
![screenshot from 2017-11-17 14-23-51](https://user-images.githubusercontent.com/1611939/32947113-0296cbd4-cba3-11e7-995e-69477f3f4392.png)